smartsim/settings/settings.py

```diff
@@ -145,15 +145,18 @@ def create_run_settings(
         "pbs": ["aprun", "mpirun", "mpiexec"],
         "cobalt": ["aprun", "mpirun", "mpiexec"],
         "lsf": ["jsrun", "mpirun", "mpiexec"],
+        "local": [""],
     }
 
     if launcher == "auto":
         launcher = detect_launcher()
 
-    def _detect_command(launcher: str):
+    def _detect_command(launcher: str) -> str:
         if launcher in by_launcher:
             for cmd in by_launcher[launcher]:
-                if is_valid_cmd(cmd):
+                if launcher == "local":
+                    return cmd
```

Comment:
I think we lost the second part of your comment, but great suggestion! Swapped to avoid calling local comparison repeatedly.

I'd really like to pull these `is_valid_cmd` checks out of `create_run_settings` completely in the future...

We run `launcher = detect_launcher()` so I was thinking that a launcher might own a mapping that stores this. That way, we never run `_detect_command` more than once. It would also mean that each launcher would own it's own collection of commands instead of letting `create_run_settings` have to know about it